### PR TITLE
Add support for RESTful WMTS

### PR DESCRIPTION
**Pull request by pomakis**
I've augmented the WebMapTileServiceImageryProvider class to support RESTful requests (by accepting a tile-URL template as defined by the WMTS 1.0.0 specification).  The tests and jsdoc have also been updated accordingly.

(This is my first github pull request ever, so if I did something wrong or am missing something, please let me know.)

### CODE REVIEW

**Comment by pjcozzi** [comment]
Wow, that was fast.  Thanks @pomakis!  Who is the right person to review this?  @kring?  @shunter?

**Comment by pjcozzi** [comment]
@pomakis can you please merge in the master branch from the main Cesium repo since there is a merge conflict (probably a trivial one in CHANGES.md):

image

File: Source/Scene/WebMapTileServiceImageryProvider.js [hunk 1]
```diff
@@ -33,13 +33,13 @@ define([
...
      * Provides tiled imagery served by {@link http://www.opengeospatial.org/standards/wmts|WMTS 1.0.0} compliant servers.
-     * This provider supports HTTP KVP-encoded GetTile requests, but does not yet support SOAP and RESTful encoding.
+     * This provider supports HTTP KVP-encoded and RESTful GetTile requests, but does not yet support the SOAP encoding.
      *
      * @alias WebMapTileServiceImageryProvider
      * @constructor
      *
      * @param {Object} options Object with the following properties:
-     * @param {String} options.url The WMTS server url.
+     * @param {String} options.url The base URL for the WMTS GetTile operation (for KVP-encoded requests) or the tile-URL template (for RESTful requests). The tile-URL template should contain the following variables: &#123;style&#125;, &#123;TileMatrixSet&#125;, &#123;TileMatrix&#125;, &#123;TileRow&#125;, &#123;TileCol&#125;. The first two are optional if actual values are hardcoded or not required by the server.
```
Comment (pjcozzi) [inline Source/Scene/WebMapTileServiceImageryProvider.js]: Do you mind expanding the `@example` below to show one that is KVP-encoded and another that is the tile-URL template?

**Comment by pjcozzi** [comment]
This looks good to me otherwise.  Does anyone else want to look at this?

**Comment by pjcozzi** [comment]
Also, please add yourself to [CONTRIBUTORS.md](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CONTRIBUTORS.md).

File: Source/Scene/WebMapTileServiceImageryProvider.js (hunk 1 above)
Comment (pomakis) [inline Source/Scene/WebMapTileServiceImageryProvider.js]: Done!  To clarify the similarities and differences between the two approaches (and to make sure the example uses a server that supports CORS), the example I added is exactly equivalent to the KVP example, but using the server's REST interface instead.

**Comment by pjcozzi** [comment]
Looks good.  Tests also look good.  Can you please merge in master from the main Cesium repo and then we can merge this?

**Comment by pjcozzi** [comment]
Thanks @pomakis.  Just FYI, github doesn't notify us if a new commit is made to a branch with an open pull request so we usually make a comment on the pull request like `This is ready` to let people know.

**Pull request merged**